src/lib/statistics/chunk_statistics/histograms/histogram_utils.hpp

```diff
@@ -0,0 +1,104 @@
+  return std::nextafter(value, -std::numeric_limits<T>::infinity());
+}
+
+template <typename T>
+std::enable_if_t<std::is_integral_v<T>, T> next_value(const T value) {
+  return value + 1;
+}
+
+template <typename T>
+std::enable_if_t<std::is_floating_point_v<T>, T> next_value(const T value) {
+  return std::nextafter(value, std::numeric_limits<T>::infinity());
+}
+
+/**
+ * Return the next representable string after `value` in the domain of strings with
+ * at most length `string_prefix_length` and the possible character set of `supported_characters`.
+ */
+std::string next_value(const std::string& value, const std::string& supported_characters,
+                       const uint32_t string_prefix_length);
+
+/**
+ * Return the next representable string after `value` in the domain of all strings with the
+ * possible character set of `supported_characters`.
+ * That is, append the first supported character to `value`.
+ */
+std::string next_value(const std::string& value, const std::string& supported_characters);
+
+/**
+ * Returns the power of base to exp.
+ * The standard library function works on doubles and is inaccurate for large numbers.
+ */
+uint64_t ipow(uint64_t base, uint64_t exp);
+
+/**
+ * Returns the number of possible strings with at most length `string_prefix_length` - 1
+ * in the possible character set of `supported_characters`.
+ */
+uint64_t base_value_for_prefix_length(const uint32_t string_prefix_length, const std::string& supported_characters);
+
+/**
+ * Returns the numerical representation of a string in the domain of all strings
+ * with at most length `string_prefix_length` and in the possible character set of `supported_characters`.
+ * The numerical representation is the number of possible strings in that domain alphabetically smaller than `value`.
+ */
+uint64_t convert_string_to_number_representation(const std::string& value, const std::string& supported_characters,
+                                                 const uint32_t string_prefix_length);
+
+/**
+ * Returns the string for a numerical representation of a string in the domain of all strings
+ * with at most length `string_prefix_length` and in the possible character set of `supported_characters`.
+ */
+std::string convert_number_representation_to_string(const uint64_t value, const std::string& supported_characters,
+                                                    const uint32_t string_prefix_length);
+
+/**
+ * Returns a pair of supported characters and prefix length.
+ * If no prefix length is supplied, it will return the maximum possible prefix length for the supported characters.
+ * If no set of supported characters is given, it will return ASCII characters 32 - 126,
+ * which are the vast majority of printable ASCII characters, and the maximum prefix length for that set (9).
+ */
+std::pair<std::string, uint32_t> get_default_or_check_string_histogram_prefix_settings(
```

Comment:
reading the doc twice I am still at loss what this function does. I can't tell from the name either, really.

I either checks whether prefix/characters are valid OR returns the default values for them? Why have this in one function? What is valid what is invalid?

Reply:
The idea is that there are basically three cases to support:
1. Neither a supported character set nor a prefix length is supplied. In that case, return the default.
2. A supported character set is supplied, but not prefix length. In that case, check the character set and return it along with the maximum prefix length supported for that set.
3. Both is supplied, so check both and their compatability.
It is not allowed to only supply a prefix length.

I added that to the docs. Do you think that helps?
The validation is in separate functions which are simply called depending on which parameters are supplied.